dotnetv3/S3/scenarios/S3ObjectLockScenario/S3ObjectLockWorkflow/S3ObjectLockWorkflow.cs

```diff
@@ -0,0 +1,436 @@
+        await _s3ActionsWrapper.CreateBucketWithLockOptions(noLockBucketName, false);
+        await _s3ActionsWrapper.CreateBucketWithLockOptions(lockEnabledBucketName, true);
+        await _s3ActionsWrapper.CreateBucketWithLockOptions(retentionAfterCreationBucketName, false);
+
+        Console.WriteLine("Press Enter to continue.");
+        if (interactive)
+            Console.ReadLine();
+
+        Console.WriteLine("\nA bucket can be configured to use object locking with a default retention period.");
+        await _s3ActionsWrapper.ModifyBucketDefaultRetention(retentionAfterCreationBucketName, true,
+            ObjectLockRetentionMode.Governance, DateTime.UtcNow.AddDays(1));
+
+        Console.WriteLine("Press Enter to continue.");
+        if (interactive)
+            Console.ReadLine();
+
+        Console.WriteLine("\nObject lock policies can also be added to existing buckets.");
+        await _s3ActionsWrapper.EnableObjectLockOnBucket(lockEnabledBucketName);
+        
+        Console.WriteLine("Press Enter to continue.");
+        if (interactive)
+            Console.ReadLine();
+
+        // Upload some files to the buckets.
+        Console.WriteLine("\nNow let's add some test files:");
+        var fileName = _configuration["exampleFileName"] ?? "exampleFile.txt";
+        int fileCount = 2;
+        // Create the file if it does not already exist.
+        if (!File.Exists(fileName))
+        {
+            await using StreamWriter sw = File.CreateText(fileName);
+            await sw.WriteLineAsync(
+                "This is a sample file for uploading to a bucket.");
+        }
+
+        foreach (var bucketName in bucketNames)
+        {
+            for (int i = 0; i < fileCount; i++)
+            {
+                var numberedFileName = Path.GetFileNameWithoutExtension(fileName) + i + Path.GetExtension(fileName);
+                fileNames.Add(numberedFileName);
+                await _s3ActionsWrapper.UploadFileAsync(bucketName, numberedFileName, fileName);
+            }
+        }
+        Console.WriteLine("Press Enter to continue.");
+        if (interactive)
+            Console.ReadLine();
+
+        if (!interactive)
+            return true;
+        Console.WriteLine("\nNow we can set some object lock policies on individual files:");
+        foreach (var bucketName in bucketNames)
+        {
+            for (int i = 0; i < fileNames.Count; i++)
+            {
+                // No modifications to the objects in the first bucket.
+                if (bucketName != bucketNames[0])
+                {
+                    var exampleFileName = fileNames[i];
+                    switch (i)
+                    {
+                        case 0:
+                            {
+                                var question =
+                                    $"\nWould you like to add a legal hold to {exampleFileName} in {bucketName}?";
```

Comment:
Add "(y/N)" to the end?

Reply:
It's added by the helper method.

Reply:
Then, update the scenario example text to show it.

Reply:
Nope, I was wrong, was looking at a different method. Fixed.